Add the setting to use the mbed-os-example-client

TomoYamanaka opened the pull request:
By default, mbed-os-example-client can't be built for Renesas GR-PEACH, and This is the workaround for the issue.

There is no device unique ID to the GR-PEACH. Therefore, we implimented the following function.
- The value read by the Timer is set to a random seed.
- Put a random number into the entropy poll, and returns it.

Regards,
Tomo Yamanaka

0xc0170 commented:
Can you please fix the formatting in the hal implementation to be compatible with mbed coding style?

Why did you decide to use Timer (c++ file) , not to use us ticker HAL functions?

bridadan commented on hal/targets.json:
```diff
@@ -1628,6 +1628,7 @@
         "extra_labels": ["RENESAS", "MBRZA1H"],
         "supported_toolchains": ["ARM", "GCC_ARM", "IAR"],
         "inherits": ["Target"],
+        "macros": [" MBEDTLS_ENTROPY_HARDWARE_ALT "],
```
Nit pick: Is there a reason the extra "space" characters (` `) are needed on both sides of `MBEDTLS_ENTROPY_HARDWARE_ALT`? Could the line be changed from:

```
        "macros": [" MBEDTLS_ENTROPY_HARDWARE_ALT "],
```

to

```
        "macros": ["MBEDTLS_ENTROPY_HARDWARE_ALT"],
```

TomoYamanaka commented on hal/targets.json:
```diff
@@ -1628,6 +1628,7 @@
         "extra_labels": ["RENESAS", "MBRZA1H"],
         "supported_toolchains": ["ARM", "GCC_ARM", "IAR"],
         "inherits": ["Target"],
+        "macros": ["MBEDTLS_ENTROPY_HARDWARE_ALT"],
```
Thank you your comments.

I deleted the unnecessary space.

TomoYamanaka commented:
@0xc0170 

Thank you for your comments.

> Can you please fix the formatting in the hal implementation to be compatible with mbed coding style?

Do you mean the current etropy_hardware_poll.cpp can't follow the following points in mbed OS 5.0 coding rules?

- Space after statements of type if, while, for, switch.
- Unix line endings are default option for files.
- Empty lines should have no trailing spaces.

Please let me confirm your intention, just in case.


> Why did you decide to use Timer (c++ file) , not to use us ticker HAL functions?

My understanding on your feedback is that the API declared in us_ticker_api.h is preferred in this case. Can I get it right?

0xc0170 commented:
> Do you mean the current etropy_hardware_poll.cpp can't follow the following points in mbed OS 5.0 coding rules?

Yes it should, mainly around ``if`` where ``{`` are on the same line.

> My understanding on your feedback is that the API declared in us_ticker_api.h is preferred in this case. Can I get it right?

Yes, it's HAL implementation and I don't see why that file needs t obe c++ file (using C++ API). What do you think?

TomoYamanaka commented:
@0xc0170 

Thank you for your explanation.
I understood very well.

I'll fix again by feeding back the contents.

TomoYamanaka commented:
@0xc0170 

I fixed the code by feedback your contents.
In "etropy_hardware_poll" function, I modified to reference the API declared in us_ticker_api.h.
And along the mbed OS 5.0 coding rules, I modified the code.

0xc0170 commented:
/morph test

0xc0170 commented:
The code looks good, thanks for updating it. we'll wait for CI results

TomoYamanaka commented:
@sg- 

Thank you for your clarification. 
Could you inform me of a little more detail schedule for "mbed OS 5.2" ?

I have the opportunity in which mbed Client is must-have function. 
If I would enable "MBEDTLS_TEST_NULL_ENTROPY", it might be impact to it.
So, I would like to clarify the schedule for "mbed OS 5.2" for further consideration.

Also, I'm not sure what the "storage driver" is. 
Could you inform me of the supplementary information on it?

Osamu-Nakamura commented:
@sg- 

I have a question on the usage of no entropy. When I tried no entropy in accordance with  https://github.com/ARMmbed/mbed-os-example-client#requirements-for-non-k64f-board, I faced a compiling error. It seems that MBEDTLS_NO_DEFAULT_ENTROPY_SOURCES isn't defined in mbed_config.h in the problem case and it caused an error. Instead, if I added MBEDTLS_NO_DEFAULT_ENTROPY_SOURCES and MBEDTLS_TEST_NULL_ENTROPY to targets.json, I could successfully build the program.

Is any additional procedure needed when modifying mbed_app.json?

sg- commented:
Both macros ```MBEDTLS_NO_DEFAULT_ENTROPY_SOURCES and MBEDTLS_TEST_NULL_ENTROPY``` should be added to the mbed_app.json file

toyowata commented:
@sg- 
I think current example of mbed_app.json for the K64F boards is not correct and created the ticket here: https://github.com/ARMmbed/mbed-os-example-client/issues/84

sg- commented:
The mbed TLS entropy integration is done via TRNG HAL. If the device supports TRNG that is the way to enable.

sg- commented:
NV_SEED didn't make it into mbed OS 5.2 Will provide more updates when available.

Osamu-Nakamura commented:
We Renesas strongly hope NV_SEED is supported in the near future update (i.e. mbed OS 5.3 or mbed OS 5.4). Otherwise, we cannot help providing users with poor security. It should be disadvantage for both you and us. Please take this point into consideration. Thank you for your understanding.
